examples/hosted-wallet-kt/pfi/src/main/kotlin/api/Exchanges.kt

```diff
@@ -150,14 +185,19 @@ class Exchanges(private val bearerDid: BearerDid, private val offeringsRepositor
         close.sign(bearerDid)
         close.verify()
 
-        val replyTo = this.exchangesToReplyTo[exchangeId] ?: throw Exception("replyTo cannot be null")
-
         println("Replying with close")
 
+        this.exchangeIdToExchange[exchangeId] = this.exchangeIdToExchange[exchangeId]!!.copy(close = close)
+
+        val replyTo = this.exchangesToReplyTo[exchangeId]
         this.replyRequest(replyTo, ReplyToRequestBody(close))
     }
 
-    private fun replyRequest(replyTo: String, body: ReplyToRequestBody) {
+    private fun replyRequest(replyTo: String?, body: ReplyToRequestBody) {
+        if (replyTo == null) {
```

Comment:
I think it may be slightly better practice to move this condition up the call stack, such that `replyRequest()` isn't called given the wallet didn't include the `replyTo` feature

slightly more verbose but more true to the intent of the function

Reply:
yup I agree!